Library/Homebrew/cask/cask_loader.rb

```diff
@@ -171,17 +174,30 @@ def self.try_new(ref, warn: false)
         new(uri)
       end
 
-      attr_reader :url
+      attr_reader :url, :name
 
       sig { params(url: T.any(URI::Generic, String)).void }
       def initialize(url)
         @url = URI(url)
-        super Cache.path/File.basename(T.must(@url.path))
+        @name = File.basename(T.must(@url.path))
+        super Cache.path/name
       end
 
       def load(config:)
         path.dirname.mkpath
 
+        match = url.to_s.match(%r{githubusercontent.com/[\w-]+/[\w-]+/[a-f0-9]{40}(?:/Casks)?/(?<name>[\w+-.@]+).rb})
+        if match
+          raise UnsupportedInstallationMethod,
+                "Installation of #{match[:name]} from a GitHub commit URL is unsupported! " \
+                "`brew extract #{match[:name]}` to a stable tap on GitHub instead."
+        elsif ALLOWED_URL_SCHEMES.exclude?(url.scheme)
```

Comment:
```suggestion
        if ALLOWED_URL_SCHEMES.exclude?(url.scheme)
```

I reckon, the above message/check is redundant now because it would be covered by the latter case anyway, right?

Reply:
Yep, good call